upickle/shared/src/test/scala/example/OptionsAsNullTests.scala

```diff
@@ -0,0 +1,82 @@
+package example
+import acyclic.file
+import upickle.{Js, TestUtil}
+import utest._
+
+import example.Simple.Thing
+
+  case class Opt(a:Option[String], b:Option[Int])
+
+
+object OptionsAsNullTests extends TestSuite{
```

Comment:
I'm not sure why no tests caught the fact that `invalid` and `read` as defined above are probably wrong. Could you add a regression test to ensure that sort of mistake doesn't happen in future?